File: packages_mjs/vault-file/src/validators.ts
import { z } from 'zod';

export const VaultHeaderSchema = z.object({
    id: z.string().uuid(),
    version: z.string().regex(/^\d+\.\d+(\.\d+)?$/), // Allow 1.0 or 1.0.0
    createdAt: z.string().datetime()
});

export const VaultMetadataSchema = z.object({
    data: z.record(z.any()).default({})
});

export const VaultPayloadSchema = z.object({
    content: z.any()
});

export const VaultFileSchema = z.object({
    header: VaultHeaderSchema,
    metadata: VaultMetadataSchema,
    payload: VaultPayloadSchema
});

export class VaultValidationError extends Error {
    constructor(message: string) {
        super(message);
        this.name = 'VaultValidationError';
    }
}

export class VaultSerializationError extends Error {
    constructor(message: string) {
        super(message);
        this.name = 'VaultSerializationError';
    }
}

File: packages_py/vault_file/src/vault_file/validators.py
from typing import Dict, Any
import uuid
import re

class VaultValidationError(Exception):
    pass

class VaultSerializationError(Exception):
    pass

def validate_vault_data(data: Dict[str, Any]) -> None:
    required_keys = ["header", "metadata", "payload"]
    for key in required_keys:
        if key not in data:
            raise VaultValidationError(f"Missing required key: {key}")
    
    validate_header(data.get("header", {}))

def validate_header(header_data: Dict[str, Any]) -> None:
    required_keys = ["id", "version", "created_at"]
    for key in required_keys:
        if key not in header_data:
            raise VaultValidationError(f"Missing required header key: {key}")
            
    # UUID Validation
    try:
        uuid.UUID(header_data["id"])
    except ValueError:
        raise VaultValidationError("Invalid UUID format")
        
    # Version Format (Simple check for now)
    if not isinstance(header_data["version"], str):
         raise VaultValidationError("Version must be a string")
